exploration/0475-overriding-extending-namespacing.md

```diff
@@ -0,0 +1,240 @@
+> functionality, while a Node application might use `import`.
+
+If an implementation supports user-installed formatters, selectors, function options,
+or expression annotations, it must also support providing "namespace" prefixes for
+each installed set of functionality.
+
+In this design, each namespace prefix is a short string and is associated with a URL.
+The URL is intended to point to some publically-available copy of the add-on library's
+registry description, for use by tooling and as a reference to users such as translators.
+
+There is no requirement that an implementation read the document at the end of the URL,
+nor validate the contents in general or against the actual functionality installed.
+
+> [!NOTE]
+> It is a good idea to use ASCII strings for namespace identifiers.
+> Remember that translators (and others) in many different languages and
+> with many different keyboards need to be able to enter the prefix.
+
+The namespace prefix is part of the `name` production.
+The prefix is limited to eight characters in length and MUST be at least two characters
+in length.
+The prefix is separated from the name by a colon (U+003A COLON).
+The namespace prefix `std` is reserved and refers to the default registry.
+
+```abnf
+name      = [namespace] name-body
+namespace = name-start *7(name-char) namespace-sep
+namespace-sep = ":"
+name-body = name-start *name-char
+```
+
+The `name` production as defined here applies to:
+
+- function (selector/formatting) names
+- option names
+- spanable names
+- expression annotation names (if approved)
+
+Examples:
+
+> Add-on function:
+>
+> ```
+> Today is {$today :icu:datetime skeleton=EEEEyMdjm}
+> ```
+>
+> Add on option:
+>
+> ```
+> Today is {$today :datetime icu:skeleton=EEEEyMdjm}
+> ```
+>
+> Add-on spannables (such as markup):
+>
+> ```
+> Today is {+html:a}{$today}{-html:a}
+> ```
+>
+> Add-on expression annotation:
+>
+> ```
+> Today is {$today :datetime @my:annotation}
+> ```
+>
+> Everything altogether all at once. This probably does not work
+> correctly, since `std:datetime` may not understand `icu:skeleton`:
+>
+> ```
+> Today is {+html:a}{$today :std:datetime icu:skeleton=EEEEyMdjm @my:annotation}{-html:a}
+> ```
+
+Users, such as developers writing messages or translators creating translations,
+are not required to type the namespace prefix in message patterns unless there
+is ambiguity in the given formatting content or in the runtime.
+However, tooling might reject or have difficulty processing values without
+the prefix being present.
+
+> For example, if an implementation is using the ICU4J library, any of the
+> following messages might be acceptable alternatives:
+>
+> ```
+> Today is {$today :datetime skeleton=EEEEMd}
+> Today is {$today :datetime icu:skeleton=EEEEMd}
+> Today is {$today :icu:datetime skeleton=EEEEMd}
+> Today is {$today :icu:datetime icu:skeleton=EEEEMd}
+> ```
+
+## Alternatives Considered
```

Comment:
An additional alternative that was discussed in Seville and should be included here is registry support for prefixes during import that may include a sigil-like trailing character. So for example the ICU skeleton support could be provided by importing it with a `"icu:"` prefix, rather than a `"icu"` prefix that's automatically followed by a `:`.

This would avoid our need to further distance the relevant rules from Nmtoken, and could achieve the same goals with a softer guidance towards some recommended separator character.

Reply:
Added this option.